src/hooks/services/common/useAuthenticationState.ts

```diff
@@ -0,0 +1,35 @@
+import { useRouter } from 'next/router';
+import { useSession } from 'next-auth/react';
+import { useEffect, useState } from 'react';
+import type { AuthenticationState } from 'types/authentication';
+import { PAGE_PATH } from 'constants/common';
+
+export const useAuthenticationState = () => {
+  const session = useSession();
+
+  const router = useRouter();
+
+  const [authenticationState, setAuthenticationState] =
+    useState<AuthenticationState>({
+      user: undefined,
+      status: 'loading',
+    });
+
+  useEffect(() => {
+    if (session.status === 'unauthenticated') {
+      void router.push(PAGE_PATH.account.login);
+      alert('인증이 필요한 페이지입니다.'); // FIXME: 문구 변경할 예정입니다.
+      return;
+    }
+
+    if (session.status === 'authenticated') {
+      setAuthenticationState({
+        update: session.update,
+        user: session.data.user,
+        status: session.status,
+      });
+    }
+  }, [session.status]);
```

Comment:
ask;

status가 'loading', 'unauthenticated', 'authenticated'로 세 가지의 값이 있어서,
'loading'에 대한 처리가 필요하진 않은지 궁금합니다!

다음과 같이 처리해도 괜찮지 않을까요?👀
```suggestion
  useEffect(() => {
    if (session.status === 'loading') return;
    
    if (session.status === 'unauthenticated') {
      void router.push(PAGE_PATH.account.login);
      alert('인증이 필요한 페이지입니다.'); // FIXME: 문구 변경할 예정입니다.
      
      return;
    }

    setAuthenticationState({
      update: session.update,
      user: session.data.user,
      status: session.status,
    });
  }, [session.status]);
```

Reply:
좋은 의견 감사합니다 :)

status가 `"unauthenticated" | "loading"` 이런 형식으로 설정되어 있으면, 얼리 리턴으로 각 케이스를 타입가드해준다고 하더라고 제대로 타입이 좁혀지지 않는 것을 확인하였습니다.(이런 경우 타입가드가 안되는지 몰랐네요..ㅎㅎ)

<img width="612" alt="image" src="https://github.com/a-daily-diary/ADD.FE/assets/77317312/4d7d2aeb-8892-4074-b371-afa78ba68c8a">

아래는 실제로 선언되어 있는 타입입니다.
<img width="559" alt="image" src="https://github.com/a-daily-diary/ADD.FE/assets/77317312/34ed2082-3fee-4c3d-a6a9-b9f03a96dbcd">

그런데 코드를 들어가서보니 onUnauthenticated 라는 메소드가 있는 것을 확인하였습니다. 해당 메소드 사용으로 코드가 더 깔끔하게 되었습니다! 다시 한번 확인할 수 있게 리뷰해주셔서 감사합니다!

작업 커밋 : ee3309a